Migration Guide for Deprecations removed since last stable release (1.22) (#5125)
flutter/website: src/docs/release/breaking-changes/1-22-deprecations.md

---

> title: Deprecated API removed after v1.22

Reviewer: Looking at the generated TOC, I don't think we need so many levels of headers. Maybe just h1 and h2. The deeper level of headers can be converted to bolded text or similar.

Author: I reduced the number of headings to h1,2,3. The third heading is for the individual class/change, which makes each of the changes in these guides linkable and accessible.

description: After reaching end of life, the following deprecated APIs were removed from Flutter.
---

## Summary

In accordance with Flutter's [Deprecation Policy][], deprecated APIs that reached end of life after the
1.22 stable release have been removed. This is the first time deprecated APIs have been removed from

> Flutter, and some of these deprecations pre-date our migration guide policy.

Reviewer: ```suggestion
Flutter, and some of these deprecations predate our migration guide policy.
```

All affected APIs have been compiled into this primary source to aid in migration. A

> [quick reference sheet] is available as well.

Reviewer: ```suggestion
[quick reference sheet][] is available as well.
```

> A [design document] and [article] are available for more context on Flutter's deprecation policy.

Reviewer: ```suggestion
A [design document][] and [article][] are available for more context on Flutter's deprecation policy.
```

[Deprecation Policy]: {{site.github}}/flutter/flutter/wiki/Tree-hygiene#deprecation
[quick reference sheet]: /go/deprecations-removed-after-1-22
[design document]: /go/flutter-deprecation-lifetime

> [article]: https://medium.com/flutter/deprecation-lifetime-in-flutter-e4d76ee738ad

Reviewer: ```suggestion
[article]: {{site.flutter-medium}}/deprecation-lifetime-in-flutter-e4d76ee738ad
```

> ## Changes

Reviewer: We need to have at least one sentence of text between headings. How about:

```
This section lists the deprecations, listed by the affected class.
```

parameter, references should be removed.

#### Migration Guide

Code before migration:
<!-- skip -->
```dart
const CupertinoTextThemeData themeData = CupertinoTextThemeData(brightness: Brightness.dark);
themeData.copyWith(brightness: Brightness.light);
```

Code after migration:
<!-- skip -->
```dart
const CupertinoTextThemeData themeData = CupertinoTextThemeData();
themeData.copyWith();
```

#### References

API documentation:
* [`CupertinoTextThemeData`][]

Relevant issues:
* [Revise CupertinoColors and CupertinoTheme for dynamic colors][]

Relevant PRs:
* Deprecated in [#41859][]
* Removed in [#72017][]

[`CupertinoTextThemeData`]: {{site.api}}/flutter/cupertino/CupertinoTextThemeData-class.html
[Revise CupertinoColors and CupertinoTheme for dynamic colors]: {{site.github}}/flutter/flutter/issues/35541
[#41859]: {{site.github}}/flutter/flutter/pull/41859
[#72017]: {{site.github}}/flutter/flutter/pull/72017

---

### Pointer Events Constructed `fromHoverEvent`

Supported by fix tool: Yes
 
The `fromHoverEvent` constructors for `PointerEnterEvent` and `PointerExitEvent` were deprecated
in v1.4.3.
The `fromMouseEvent` constructor should be used instead.
 

> #### Migration Guide

Reviewer: => #### Migration guide

(We use sentence case headers.)

final PointerExitEvent exitEvent = PointerExitEvent.fromHoverEvent(PointerHoverEvent());
```
 
Code after migration:
<!-- skip -->
```dart
final PointerEnterEvent enterEvent = PointerEnterEvent.fromMouseEvent(PointerHoverEvent());
final PointerExitEvent exitEvent = PointerExitEvent.fromMouseEvent(PointerHoverEvent());
```
 
#### References
 
API documentation:
* [`PointerEnterEvent`][]
* [`PointerExitEvent`][]
 
Relevant issues:
* [PointerEnterEvent and PointerExitEvent can only be created from hover events][]
 
Relevant PRs:
* Deprecated in [#28602][]
* Removed in [#72395][]
 
[`PointerEnterEvent`]: {{site.api}}/flutter/gestures/PointerEnterEvent-class.html
[`PointerExitEvent`]: {{site.api}}/flutter/gestures/PointerExitEvent-class.html
[PointerEnterEvent and PointerExitEvent can only be created from hover events]: {{site.github}}/flutter/flutter/issues/29696
[#28602]: {{site.github}}/flutter/flutter/pull/28602
[#72395]: {{site.github}}/flutter/flutter/pull/72395

---

### `showDialog` uses `builder`

Supported by fix tool: Yes
 
The `child` parameter of `showDialog` was deprecated in in v0.2.3.
The `builder` parameter should be used instead.
 

> #### Migration Guide

Reviewer: => Migration guide

Global change.

`WidgetsFlutterBinding.ensureInitialized` was called. Instead, replace calls to this method with
`RenderView.prepareInitialFrame`, followed by `RenderView.owner.requestVisualUpdate`.
 
#### Migration Guide
 
Code before migration:
<!-- skip -->
```dart
scheduleInitialFrame();
```
 
Code after migration:
<!-- skip -->
```dart
prepareInitialFrame();	
owner.requestVisualUpdate();
```
 
#### References
 
API documentation:
* [`RenderView`][]
* [`WidgetsFlutterBinding`][]
 
Relevant issues:
* [WidgetsFlutterBinding.ensureInitialized() takes down splash screen too early][]

Relevant PRs:
* Deprecated in [#39535][]
* Removed in [#73748][]
 
[`RenderView`]: {{site.api}}/flutter/rendering/RenderView-class.html
[`TextSpan`]: {{site.api}}/flutter/widgets/WidgetsFlutterBinding-class.html
[WidgetsFlutterBinding.ensureInitialized() takes down splash screen too early]: {{site.github}}/flutter/flutter/issues/39494
[#39535]: {{site.github}}/flutter/flutter/pull/39535
[#73748]: {{site.github}}/flutter/flutter/pull/73748

---

### `Layer.findAll`

Supported by fix tool: No

The `Layer.findAll` method was deprecated with the introduction of `Layer.findAnnotations` in order

> to unify the implementations of `find` and `findAll`. To migrate affected code, call on

Reviewer 2: Nit: "call on" => "call"

If I'm reading this correctly.

Reviewer: +1

`BinaryMessages.setMockMessageHandler` | `ServicesBinding.instance.defaultBinaryMessenger.setMockMessageHandler`
 
#### References
 
API documentation:
* [`ServicesBinding`][]
* [`BinaryMessenger`][]
 
Relevant issues:
* [Flutter synchronization support for Espresso/EarlGrey][]

Relevant PRs:
* Initially changed in [#37489][]
* Deprecated in [#38464][]
* Removed in [#73750][]
 
[`ServicesBinding`]: {{site.api}}/flutter/services/ServicesBinding-mixin.html
[`BinaryMessenger`]: {{site.api}}/flutter/services/BinaryMessenger-class.html
[Flutter synchronization support for Espresso/EarlGrey]: {{site.github}}/flutter/flutter/issues/37409
[#37489]: {{site.github}}/flutter/flutter/pull/37489
[#38464]: {{site.github}}/flutter/flutter/pull/38464
[#73750]: {{site.github}}/flutter/flutter/pull/73750

---

### Generic methods for `BuildContext`

Supported by fix tool: Yes

Several methods in `BuildContext` were using `Type` to search for ancestors. Most of those methods
implied a cast at call site because their return type was a parent type. Moreover the type provided
was not checked at analysis time even if the type is actually constrained. Making these methods

> generics improves type safety and need less code.

Reviewer 2: Nit: "need less code" => "requires less code"

Or maybe these were copied from somewhere and shouldn't change.

Reviewer: +1

Supported by fix tool: No

The `WaitUntilNoTransientCallbacks`, `WaitUntilNoPendingFrame`, and `WaitUntilFirstFrameRasterized`
methods from the `flutter_driver` packages were deprecated and removed in order to provide a more
composable `waitForCondition` API that can be used to compose conditions that the client would like
to wait for.
 
#### Migration Guide
 
Code before migration: | Code after migration:
-- | --
`WaitUntilNoTransientCallbacks` | `WaitForCondition(NoTransientCallbacks())`
`WaitUntilNoPendingFrame` | `WaitForCondition(NoPendingFrame())`
`WaitUntilFirstFrameRasterized` | `WaitForCondition(FirstFrameRasterized))`
 
#### References
 
API documentation:
* [`WaitForCondition`][]

Relevant issues:
* [Flutter synchronization support for Espresso/EarlGrey][]
 
Relevant PRs:
* Initially changed in [#37736][]
* Deprecated in [#38836][]
* Removed in [#73754][]
 
[`WaitForCondition`]: {{site.api}}/flutter/flutter_driver/WaitForCondition-class.html
[#37736]: {{site.github}}/flutter/flutter/pull/37736
[#38836]: {{site.github}}/flutter/flutter/pull/38836
[#73754]: {{site.github}}/flutter/flutter/pull/73754

---

> 

Reviewer 2: Nit: Too much whitespace?

---
title: Deprecated API removed after v1.22
description: After reaching end of life, the following deprecated APIs were removed from Flutter.
---

## Summary

In accordance with Flutter's [Deprecation Policy][], deprecated APIs that reached end of life after the
1.22 stable release have been removed. This is the first time deprecated APIs have been removed from
Flutter, and some of these deprecations predate our migration guide policy.

All affected APIs have been compiled into this primary source to aid in migration. A
[quick reference sheet][] is available as well.

A [design document][] and [article][] are available for more context on Flutter's deprecation policy.

[Deprecation Policy]: {{site.github}}/flutter/flutter/wiki/Tree-hygiene#deprecation
[quick reference sheet]: /go/deprecations-removed-after-1-22

> [design document]: /go/flutter-deprecation-lifetime

Reviewer: Kate, this link doesn't work and has broken the build. Please fix...

Author: ```suggestion
[design document]: /go/deprecation-lifetime
```
